api/src/main/java/org/apache/apex/api/Launcher.java

```diff
@@ -77,8 +77,30 @@ public LaunchMode(Class<L> clazz)
     KILL
   }
 
-  // Marker interface
-  public interface AppHandle {}
+  /**
+   * Results of application launch. The client can interact with the running application through this handle.
+   */
+  public interface AppHandle
+  {
+    boolean isFinished();
+
+    /**
+     * Shutdown the application.
+     *
+     * The method takes the application handle and a shutdown mode. The shutdown mode specifies how to shutdown the
+     * application.
+     *
+     * If the mode is AWAIT_TERMINATION, an attempt should be made to shutdown the application in an orderly fashion
+     * and wait till termination. If the application does not terminate in a reasonable amount of time the
+     * implementation can forcibly terminate the application.
+     *
+     * If the mode is KILL, the application can be killed immediately.
+     *
+     * @param shutdownMode The shutdown mode
+     */
+    void shutdown(ShutdownMode shutdownMode) throws LauncherException;
```

Comment:
After launch you have a reference which is the handle. You can use it to interact with the app instead of having to also pass around a launcher.